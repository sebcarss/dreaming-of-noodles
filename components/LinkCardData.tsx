export interface LinkCardData {
    country: string;
    region: string;
    gridLinkImage: string;
    gridLinkUrl: string;
}