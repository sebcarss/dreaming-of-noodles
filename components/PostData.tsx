export interface PostData {
    title: string;
    id: string;
    date: Date;
    tags: string[];
    published: boolean;
    excerpt: string;
    image: string;
};