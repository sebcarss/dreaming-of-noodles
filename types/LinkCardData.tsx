export interface LinkCardData {
    frontmatter: { [key: string]: any };
    content: string;
}