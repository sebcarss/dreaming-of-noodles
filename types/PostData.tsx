export interface PostData {
    title: string;
    id: string;
    date: string;
    tags: string[];
    published: boolean;
    excerpt: string;
    image: string;
};