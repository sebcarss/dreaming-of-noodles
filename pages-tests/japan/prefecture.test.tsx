import Prefecture from "../../pages/japan/[prefecture]";
import renderer from "react-test-renderer";
import { LinkCardData } from "../../types/LinkCardData";
import { PostData } from '../../types/PostData';

const regionData: LinkCardData = {
    frontmatter: {
        country: 'Japan',
        region: 'Hokkaido',
        gridLinkUrl: '/japan/hokkaido',
        gridLinkImage: '/japan-hokkaido.jpg'
    },
    content: "\nWelcome to Hokkaido!"
};

const posts: PostData[] = [
    {
        title: "Hello World 2",
        id: "2019-01-02-hello-world-2",
        date: "2019-01-02",
        tags: ["hello", "world"],
        published: true,
        excerpt: 'Hello excerpt',
        image: "https://source.unsplash.com/random/300x200",
    }
];

describe("Prefecture page", () => {
    it('should match the snapshot', () => {
        const tree = renderer.create(<Prefecture regionData={regionData} posts={posts} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});