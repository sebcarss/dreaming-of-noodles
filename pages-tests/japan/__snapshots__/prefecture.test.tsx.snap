// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prefecture page should match the snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <nav
      className="color-nav navbar navbar-expand-lg navbar-dark fixed-top"
      style={
        {
          "boxShadow": "0 10px 10px rgba(0,0,0,.2)",
          "color": "black",
          "fontSize": "1rem",
          "textAlign": "center",
          "width": "auto",
        }
      }
    >
      <a
        className="mx-2 navbar-brand"
        href="/"
      >
        Dreaming of Noodles
      </a>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          className="me-auto navbar-nav"
          onKeyDown={[Function]}
          style={
            {
              "textAlign": "left",
            }
          }
        >
          <a
            className="nav-link"
            data-rb-event-key="/japan"
            href="/japan"
            onClick={[Function]}
            onKeyDown={[Function]}
            style={
              {
                "color": "#db5004",
              }
            }
          >
            Japan
          </a>
          <a
            className="nav-link"
            data-rb-event-key="/recipe-index"
            href="/recipe-index"
            onClick={[Function]}
            onKeyDown={[Function]}
            style={
              {
                "color": "#db5004",
              }
            }
          >
            Recipe Index
          </a>
        </div>
      </div>
    </nav>
    <main
      className="mt-5"
    >
      <h1
        className="mt-3"
      >
        Hokkaido
      </h1>
      <div
        className="row"
      >
        <div>
          <div
            className="col-md-6 col-12"
            style={
              {
                "float": "left",
                "marginRight": "1rem",
              }
            }
          >
            <span
              style={
                {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "inline-block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "overflow": "hidden",
                  "padding": 0,
                  "position": "relative",
                  "width": "initial",
                }
              }
            >
              <span
                style={
                  {
                    "background": "none",
                    "border": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "maxWidth": "100%",
                    "opacity": 1,
                    "padding": 0,
                    "width": "initial",
                  }
                }
              >
                <img
                  alt=""
                  aria-hidden={true}
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27640%27%20height=%27640%27/%3e"
                  style={
                    {
                      "background": "none",
                      "border": 0,
                      "display": "block",
                      "height": "initial",
                      "margin": 0,
                      "maxWidth": "100%",
                      "opacity": 1,
                      "padding": 0,
                      "width": "initial",
                    }
                  }
                />
              </span>
              <img
                data-nimg="intrinsic"
                decoding="async"
                onError={[Function]}
                onLoad={[Function]}
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style={
                  {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
              <noscript>
                <img
                  data-nimg="intrinsic"
                  decoding="async"
                  loading="lazy"
                  src="/_next/image?url=https%3A%2F%2Fdreamingofnoodles.s3.eu-west-1.amazonaws.com%2Fimages%2Fundefined&w=1920&q=75"
                  srcSet="/_next/image?url=https%3A%2F%2Fdreamingofnoodles.s3.eu-west-1.amazonaws.com%2Fimages%2Fundefined&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fdreamingofnoodles.s3.eu-west-1.amazonaws.com%2Fimages%2Fundefined&w=1920&q=75 2x"
                  style={
                    {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": undefined,
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 0,
                    }
                  }
                />
              </noscript>
            </span>
          </div>
          <div
            className="col"
            dangerouslySetInnerHTML={
              {
                "__html": "
Welcome to Hokkaido!",
              }
            }
          />
        </div>
      </div>
    </main>
  </div>
  <footer
    className="mb-3"
  >
    <hr />
    <div
      className="lead text-center"
    >
      Recipes by Seb Carss
    </div>
  </footer>
</div>
`;
